web/base_viewer.js

```diff
@@ -1336,6 +1336,7 @@ class BaseViewer {
       return {
         width: viewport.width,
         height: viewport.height,
+        scale: pageView.scale,
```

Comment:
The `pageView.scale` property simply contains the *current* zoom-factor used in the viewer itself, hence I don't really understand how this is relevant for printing!?

Unless I'm completely misremembering/misunderstanding things printing were supposed to be *separate* from the rendering in the viewer, and having the printing output connected to the current viewer rendering thus seem strange.

Reply:
Hmm, so this does fix it, so something is going wrong. Maybe Firefox is copying the zoom factor over to the printed page, let me check.

Reply:
Alright, so yeah, this is not the right thing to do.

This patch does fix it at the "automatic zoom level" of course, but the behavior of course starts depending on the zoom factor which is obviously broken.

Do you know where the automatic zoom level is being computed?

Reply:
> Do you know where the automatic zoom level is being computed?

I'm not sure how helpful it is, but see below; as mentioned it seems strange to connect the printing output to the current viewer state (since they should be independent), so it'd be good to understand *exactly* what goes wrong in the current code.
https://github.com/mozilla/pdf.js/blob/7fef896a9fba6366b22da70747d8eff273ba47e9/web/base_viewer.js#L791-L798

Reply:
Yes, agreed. I'll split my other fix in another PR, since they should be independent.

Reply:
Ok, so I see what's going wrong. The issue is that the PDF is for an A5 page, rather than A4, but Firefox doesn't support `@page { size }` so pdf.js can't set the right page size for the user.

Reply:
So this is invalid, afaict, #13100 is still right though.